src/features/device-state/state-patch-utils.ts

```diff
@@ -39,6 +39,46 @@ export const v2ValidPatchFields: Array<
 	'download_progress',
 ];
 
+const SHORT_TEXT_LENGTH = 255;
+const ADDRESS_DELIMITER = ' ';
+
+// Truncate text at delimiters to input length or less
+const truncateText = (
+	longText: string,
+	length: number = SHORT_TEXT_LENGTH,
+	delimiter: string = ADDRESS_DELIMITER,
+): string => {
+	return longText
+		.split(delimiter)
+		.reduce((text, fragment) => {
+			const textWithFragment = text + delimiter + fragment;
+			return textWithFragment.length <= length ? textWithFragment : text;
+		}, '')
+		.trim();
+};
+
+type ValidPatchFields = Array<
+	typeof v3ValidPatchFields[number] | typeof v2ValidPatchFields[number]
+>;
+
+export const truncateShortTextFields = (
+	object: Dictionary<any>,
+	keysToTruncate: ValidPatchFields,
+) => {
+	keysToTruncate.forEach((key) => {
+		if (
+			typeof object[key] !== 'string' ||
+			object[key].length <= SHORT_TEXT_LENGTH
+		) {
+			return;
+		}
+		if (['ip_address', 'mac_address'].includes(key)) {
+			object[key] = truncateText(object[key]);
+		}
```

Comment:
@fisehara @cywang117 I guess there is no need to have both the `keysToTruncate` & the `['ip_address', 'mac_address']` here, right?
B/c it look like an unnecessary safeguard on whether the invoker used shorttext props or not.
Given how the source looks today, maybe we could just do:
```suggestion
const default_short_text_fields_to_trancate = ['ip_address', 'mac_address'];
export const truncateShortTextFields = (
	object: Dictionary<any>,
	keysToTruncate: ValidPatchFields = default_short_text_fields_to_trancate,
) => {
	for (const key of keysToTruncate) {
		if (
			typeof object[key] !== 'string' ||
			object[key].length <= SHORT_TEXT_LENGTH
		) {
			return;
		}
		object[key] = truncateText(object[key]);
```
so that this array doesn't have to get repeated 3 times and we have a single definition. Less GC as well :)
WRT the `for...of`, Page recently replaced tons of `.forEach` b/c it's slightly faster, so let's do it here as well for consistency.

Reply:
See https://github.com/balena-io/open-balena-api/pull/1085 for the application of the changes suggested here.